presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -696,22 +697,28 @@ protected ExtractionResult visitInPredicate(InPredicate node, Boolean complement
 
             InListExpression valueList = (InListExpression) node.getValueList();
             checkState(!valueList.getValues().isEmpty(), "InListExpression should never be empty");
-
-            ImmutableList.Builder<Expression> disjuncts = ImmutableList.builder();
-            for (Expression expression : valueList.getValues()) {
-                disjuncts.add(new ComparisonExpression(EQUAL, node.getValue(), expression));
+            Iterator<Expression> expressionIterator = valueList.getValues().iterator();
+            TupleDomain<Symbol> tupleDomain = process(new ComparisonExpression(EQUAL, node.getValue(), expressionIterator.next()), complement)
+                    .getTupleDomain();
+            while (expressionIterator.hasNext() && !tupleDomain.isAll()) {
```

Comment:
`!tupleDomain.isAll()` does not sound correct for complement here. I hope I am wrong otherwise it is very sad that tests didn't catch that.

Reply:
For complement it doesn't matter right ? `!tupleDomain.isAll()` is used to break the iterations irrespective of complement